dojo/tasks.py

```diff
@@ -75,6 +75,24 @@ def add_alerts(self, runinterval):
             calculate_grade(product)
 
 
+@app.task(bind=True)
+def cleanup_alerts(*args, **kwargs):
+    try:
+        max_alerts_per_user = settings.MAX_ALERTS_PER_USER
+    except System_Settings.DoesNotExist:
+        max_alerts_per_user = -1
+
+    if max_alerts_per_user > -1:
+        total_deleted_count = 0
+        logger.info('start deleting oldest alerts if a user has more than %s alerts', max_alerts_per_user)
+        users = User.objects.all()
+        for user in users:
+            alerts_to_delete = Alerts.objects.filter(user_id=user.id).order_by('-created')[max_alerts_per_user:].values_list("id", flat=True)
```

Comment:
Would not this actually return the top most recent `x` alerts, due to the `-` in front of `created` in the `order_by` sorting descending?

Reply:
Yes, but it start deleting only after the first bunch due to the [:]

Reply:
[:] syntax is tricky sometime :D but the variable name is very clear. (and the end of the instruction `.values_list("id", flat=True)` clearly show what is returned)